operators/pkg/examagent/instance.go

```diff
@@ -0,0 +1,289 @@
+// Copyright 2020-2022 Politecnico di Torino
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package examagent contains the main logic and helpers
+// for the crownlabs exam agent component.
+package examagent
+
+import (
+	"bytes"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	clv1alpha2 "github.com/netgroup-polito/CrownLabs/operators/api/v1alpha2"
+	"github.com/netgroup-polito/CrownLabs/operators/pkg/forge"
+)
+
+//go:embed redirecting.html
+var httpPageStartingUp string
+
+// InstanceAdapter represents an Instance within the examagent.
+type InstanceAdapter struct {
+	ID                string                               `json:"id"`
+	Template          string                               `json:"template"`
+	Running           *bool                                `json:"running,omitempty"`
+	CustomizationUrls clv1alpha2.InstanceCustomizationUrls `json:"customizationUrls"`
+}
+
+// InstanceHandler is the handler for the InstanceAdapter.
+type InstanceHandler struct {
+	Log    logr.Logger
+	Client client.Client
+}
+
+const (
+	// InstanceAdapterEndpoint is the endpoint for the InstanceAdapter.
+	InstanceAdapterEndpoint = "/instance"
+
+	// XForwardedFor -> "X-Forwarded-For" header.
+	XForwardedFor = "X-Forwarded-For"
+)
+
+// ServeHTTP is the Instance handler for the examagent.
+func (ih InstanceHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
```suggestion
func (ih *InstanceHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

I suppose. In alternative, is this is requested so, I would change the others for consistency.

Reply:
It is required to be this way, still using the pointer where it's possible isn't it better in terms of performance? I don't really know how big are the objects inside...